### PR TITLE
Fix translations not loading for Javascript files

### LINKED ISSUE #1712: Lang pack translations are not being loaded for JS files
After changing the Site Language and downloading the new translations, only the PHP string are being translated.

**Steps to reproduce:**
1. Change the site language to one that has a language pack available
2. Go to Dashboard > Updates, at the bottom of the page click: Update Translations
3. Go to Payments, and see that not all texts are in the site language (ex: table headers)

### PR DESCRIPTION
**Pull request by diegocurbelo**
Fixes #1712 and #1715

### Changes proposed in this Pull Request

A language pack zip file contains a `json file` for each `js file path` present in the original POT file, and that json file has the md5 hash of the file path in its name (ex: `woocommerce-payments-es_ES-c817bb653276f1ca458c82d18b93b92a.json`), but the UI uses bundled files (dist/index.js) so, no json file is loaded for them.

This PR ports the solution used in the `woocommerce-admin` plugin, which consists of combining all the json chunk files into a single `woocommerce-payments-es_ES.json` file after getting new translations (using the action `upgrader_process_complete`), and adds a filter (`load_script_translation_file`) to load it for all `WCPAY_*` script handles.

__NOTE__: The table footer (summary) not translated (#1715) is also fixed, to test it add the following `messages` to the `woocommerce-payments-es_ES.json` file (or similar depending on the locale): `deposits`, `total`, `transactions`, `fees` and `net`.
This is necessary because those `strings` are `new` and new translation strings are uploaded to _GlotPress_ during the release process, so they are not included in the language pack... (they will be included after merging to develop and creating a test release)

### Testing instructions

1. Change the site language to one that has a language pack available (ex: Español)
2. Go to Dashboard > Updates, at the bottom of the page click: `Update Translations`
3. Go to Payments (ex: Pagos), and see that all texts are translated to the selected site language

Before:
Screen Shot 2021-05-04 at 02 25 12

After:
Screen Shot 2021-05-04 at 02 24 16

-------------------

- [ ] Added changelog entry (or does not apply)
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

File: includes/class-wc-payments.php
Comment (reykjalin) [reply includes/class-wc-payments.php:751]: Maybe it'd be better to add a note here that this funciton should only be called during the `upgrader_process_complete` action?

File: includes/class-wc-payments.php
Comment (reykjalin) [inline includes/class-wc-payments.php:832]: I think a better check here would be `0 !== strpos( $handle, 'WCPAY_' )` to make sure we're checking for the prefix 🙂 . At least I think all our scripts start with `WCPAY_` otherwise that wouldn't really work 😅 

The way this currently works any script with `WCPAY_` anywhere (+ the `woocommerce-payments` domain) would load the translation files 🙂. I don't really think that's a problem, but it doesn't hurt to be a bit more precise here I think ☺️

File: includes/class-wc-payments.php
Comment (reykjalin) [inline includes/class-wc-payments.php:230]: I wonder if it'd be better to create a class for this custom translation handling instead of adding more things to `class-wc-payments.php`. I don't even think we'd need to add a class, maybe it's ok to just `require_once` a file that sets this up?

I think having some sort of `class-wc-payments-translation-loader.php` (or something) would help with the maintenance of this going forward 🙂 .

**Review by reykjalin — CHANGES_REQUESTED** [review]
This tests really well! Works flawlessly 🥇 

I had a couple of comments — mostly just nitpicks — but I think they're something we should address before merging ☺️

File: includes/class-wc-payments.php
Comment (diegocurbelo) [reply includes/class-wc-payments.php:751]: > Is this assumption something we should worry about? What happens when WP_Filesystem is not initialized with write access?

This function triggers after the translations are updated, and that wouldn't be possible if the `WP_Filesystem is not initialized with write access` so I don't think we should worry.

> Maybe it'd be better to add a note here that this funciton should only be called during the upgrader_process_complete action?

There is a comment saying that but is inside the function, I'll move it to the function doc.

File: includes/class-wc-payments.php
Comment (diegocurbelo) [reply includes/class-wc-payments.php:832]: Good catch!

File: includes/class-wc-payments.php
Comment (diegocurbelo) [reply includes/class-wc-payments.php:230]: I agree it would help long term maintenance, I'll move the code to its own class 😄

File: includes/class-wc-payments.php
Comment (diegocurbelo) [reply includes/class-wc-payments.php:706]: I was a bit reluctant to add that constant as new developers might use it as the domain for translations (`__( 'text', Class::SLUG )`). But we can add a `Note` to the constant declaration, and once support for php 7.0 is dropped, we can change it to a private constant.

Comment (reykjalin) [reply includes/class-wc-payments.php]: That makes sense, thanks!

File: includes/class-wc-payments.php
Comment (reykjalin) [reply includes/class-wc-payments.php]: >I'll move [the comment] to the function doc.

Thank you! That should make it visibile when people hover over the function which we'll say is visible enough 😛

**Review by reykjalin — APPROVED** [review]
Looks great and tests well! 🥇 

Let's merge 🎉 🚀

**Pull request merged**
